// @clevercloud/client/cjs/api/v4/application.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.streamAccessLogs = streamAccessLogs;
exports.streamLogs = streamLogs;

var _pickNonNull = require("../../pick-non-null.js");

/**
 * GET /logs/organisations/{ownerId}/applications/{applicationId}/accesslogs
 * @param {Object} params
 * @param {String} params.undefined
 * @param {String} params.undefined
 * @param {String} params.since
 * @param {String} params.until
 * @param {String} params.limit
 * @param {String} params.field
 * @param {String} params.throttleElements
 * @param {String} params.throttlePerInMilliseconds
 * @param {String} params.Last-Event-ID
 */
function streamAccessLogs(params) {
  // no multipath for /self or /organisations/{id}
  return Promise.resolve({
    method: 'get',
    url: `/v4/logs/organisations/${params.ownerId}/applications/${params.applicationId}/accesslogs`,
    headers: {
      Accept: 'text/event-stream'
    },
    queryParams: (0, _pickNonNull.pickNonNull)(params, ['since', 'until', 'limit', 'field', 'throttleElements', 'throttlePerInMilliseconds']) // no body

  });
}
/**
 * GET /logs/organisations/{ownerId}/applications/{applicationId}/logs
 * @param {Object} params
 * @param {String} params.undefined
 * @param {String} params.undefined
 * @param {String} params.since
 * @param {String} params.until
 * @param {String} params.limit
 * @param {String} params.deploymentId
 * @param {String} params.instanceId
 * @param {String} params.filter
 * @param {String} params.service
 * @param {String} params.field
 * @param {String} params.throttleElements
 * @param {String} params.throttlePerInMilliseconds
 * @param {String} params.Last-Event-ID
 */


function streamLogs(params) {
  // no multipath for /self or /organisations/{id}
  return Promise.resolve({
    method: 'get',
    url: `/v4/logs/organisations/${params.ownerId}/applications/${params.applicationId}/logs`,
    headers: {
      Accept: 'text/event-stream'
    },
    queryParams: (0, _pickNonNull.pickNonNull)(params, ['since', 'until', 'limit', 'deploymentId', 'instanceId', 'filter', 'service', 'field', 'throttleElements', 'throttlePerInMilliseconds']) // no body

  });
}
