// @clevercloud/client/cjs/api/v4/instance.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getApplicationInstances = getApplicationInstances;
exports.getInstance = getInstance;

var _pickNonNull = require("../../pick-non-null.js");

/**
 * GET /orchestration/organisations/{ownerId}/applications/{applicationId}/instances
 * @param {Object} params
 * @param {String} params.undefined
 * @param {String} params.undefined
 * @param {String} params.since
 * @param {String} params.until
 * @param {String} params.includeState
 * @param {String} params.excludeState
 * @param {String} params.undefined
 * @param {String} params.limit
 * @param {String} params.order
 */
function getApplicationInstances(params) {
  // no multipath for /self or /organisations/{id}
  return Promise.resolve({
    method: 'get',
    url: `/v4/orchestration/organisations/${params.ownerId}/applications/${params.applicationId}/instances`,
    headers: {
      Accept: 'application/json'
    },
    queryParams: (0, _pickNonNull.pickNonNull)(params, ['since', 'until', 'includeState', 'excludeState', 'limit', 'order']) // no body

  });
}
/**
 * GET /orchestration/organisations/{ownerId}/applications/{applicationId}/instances/{instanceId}
 * @param {Object} params
 * @param {String} params.undefined
 * @param {String} params.undefined
 * @param {String} params.undefined
 */


function getInstance(params) {
  // no multipath for /self or /organisations/{id}
  return Promise.resolve({
    method: 'get',
    url: `/v4/orchestration/organisations/${params.ownerId}/applications/${params.applicationId}/instances/${params.instanceId}`,
    headers: {
      Accept: 'application/json'
    } // no query params
    // no body

  });
}
