// @clevercloud/client/cjs/api/v4/product.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getAllZones = getAllZones;
exports.listZones = listZones;
exports.getZone = getZone;

var _pickNonNull = require("../../pick-non-null.js");

/**
 * GET /products/zones
 * @param {Object} params
 */
function getAllZones() {
  // no multipath for /self or /organisations/{id}
  return Promise.resolve({
    method: 'get',
    url: `/v4/products/zones`,
    headers: {
      Accept: 'application/json'
    } // no query params
    // no body

  });
}
/**
 * GET /products/zones
 * @param {Object} params
 * @param {String} params.undefined
 */


function listZones(params) {
  // no multipath for /self or /organisations/{id}
  return Promise.resolve({
    method: 'get',
    url: `/v4/products/zones`,
    headers: {
      Accept: 'application/json'
    } // no query params
    // no body

  });
}
/**
 * GET /products/zones/{zoneName}
 * @param {Object} params
 * @param {String} params.undefined
 */


function getZone(params) {
  // no multipath for /self or /organisations/{id}
  return Promise.resolve({
    method: 'get',
    url: `/v4/products/zones/${params.zoneName}`,
    headers: {
      Accept: 'application/json'
    } // no query params
    // no body

  });
}
