// @clevercloud/client/cjs/login.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getOauthParams = getOauthParams;
exports.fetchRequestToken = fetchRequestToken;
exports.fetchAccessToken = fetchAccessToken;

var _oauth = _interopRequireDefault(require("oauth-1.0a"));

var _oauth2 = require("./api/v2/oauth.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getOauthParams({
  consumerKey,
  consumerSecret,
  tokenSecret = ''
}) {
  // We need this for getNonce()
  _oauth.default.prototype.nonce_length = 32;
  return {
    oauth_consumer_key: consumerKey,
    oauth_signature_method: 'PLAINTEXT',
    oauth_signature: consumerSecret + '&' + tokenSecret,
    oauth_timestamp: _oauth.default.prototype.getTimeStamp(),
    oauth_nonce: _oauth.default.prototype.getNonce()
  };
}

function fetchRequestToken({
  consumerKey,
  consumerSecret,
  oauthCallback
}) {
  return (0, _oauth2.fetchRequestToken)(null, { ...getOauthParams({
      consumerKey,
      consumerSecret
    }),
    oauth_callback: oauthCallback
  });
}

function fetchAccessToken({
  consumerKey,
  consumerSecret,
  tokenSecret,
  oauthToken,
  oauthVerifier
}) {
  return (0, _oauth2.fetchAccessToken)(null, { ...getOauthParams({
      consumerKey,
      consumerSecret,
      tokenSecret
    }),
    oauth_token: oauthToken,
    oauth_verifier: oauthVerifier
  });
}
